src/tech/v3/libs/smile/discrete_nb.clj

```diff
@@ -44,9 +44,9 @@
                       thawed-model
                       model]
   "Predict function for discrete naive bayes"
-  (let [sparse-arrays (get feature-ds :bow-sparse)
+  (let [sparse-arrays (get feature-ds  (get-in model [:options :sparse-column]))
         target-colum (first (:target-columns model))
-        predictions (map #(.predict (:model-data model) %) sparse-arrays)
```

Comment:
Was this even a bug ?
Should I use the `thawed model` or the (:model-data model) here ? Bot have the model instance, I believe.

Reply:
model-data should be the serialized byte-array of the model.  Anything else will break saving a model to nippy and loading it again.  thawed-model is the result of calling the model-specific thaw function on this byte array of data.

This is important because Smile uses java serialization which is [always a security risk](https://owasp.org/www-community/vulnerabilities/Deserialization_of_untrusted_data) and thus nippy refuses to use java serialization any longer.  We need to respect that decision and ensure that the return value of 'train' does not require java serialization in order to save to nippy.

You want to always use `:thawed-model`  :-).

Reply:
Ok, the current code uses the thawed-model.
I will check the other new model I implemented, and eventually make new PRs